fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java

```diff
@@ -1248,10 +1248,15 @@ private void handleInsertStmt() throws Exception {
 
                 coord.exec();
 
-                coord.join(context.getSessionVariable().getQueryTimeoutS());
+                boolean notTimeout = coord.join(context.getSessionVariable().getQueryTimeoutS());
                 if (!coord.isDone()) {
                     coord.cancel();
-                    ErrorReport.reportDdlException(ErrorCode.ERR_EXECUTE_TIMEOUT);
+                    if (notTimeout) {
+                        ErrorReport.reportDdlException(ErrorCode.ERR_UNHEALTHY_BACKEND_EXISTS);
```

Comment:
Using query state is better

Reply:
Or can you print out the wrong backend id through the error code

Reply:
> Or can you print out the wrong backend id through the error code

OK, thanks for your review.